Define endorseTimeout value in fabric-sdk-node v1.4(Issue #552)

takayuki-nagai opened the pull request:
In fabric-sdk-node v1.4, as well as in v2.2, define "endorseTimeout" value and use it instead of commitTimeout.

Co-Authored-by: Shinsuke Hasegawa <shinsuke.hasegawa.fc@hitachi.com>
Signed-off-by: takayuki-nagai <takayuki.nagai.nu@hitachi.com>

Closes #552

takayuki-nagai committed:
Define endorseTimeout value in fabric-sdk-node v1.4(Issue #552)

In fabric-sdk-node v1.4, as well as in v2.2, define "endorseTimeout" value and use it instead of "commitTimeout".

Co-Authored-by: Shinsuke Hasegawa <shinsuke.hasegawa.fc@hitachi.com>
Signed-off-by: takayuki-nagai <takayuki.nagai.nu@hitachi.com>

bestbeforetoday commented on fabric-network/test/transaction.js:
```diff
@@ -301,15 +301,15 @@ describe('Transaction', () => {
 
 		it('sends proposal with short timeout', async () => {
-			stubContract.getEventHandlerOptions.returns({commitTimeout: 3});
+			stubContract.getEventHandlerOptions.returns({endorseTimeout: 3});
```
This test seems intended to ensure the specified timeout did not drop below the commitTimeout value. The code no longer behaves this way, and I think your new behaviour is reasonable, so I suggest just removing this test entirely. The behaviour it is testing is already covered by the previous unit test.

bestbeforetoday commented on fabric-network/test/transaction.js:
```diff
@@ -301,15 +301,15 @@ describe('Transaction', () => {
 		});
 
 		it('sends proposal with long timeout', async () => {
```
With the new behaviour, this test might be better named.
```suggestion
		it('uses endorseTimeout option as proposal timeout', async () => {
```
```suggestion
		it('sends proposal with long timeout', async () => {
```

bestbeforetoday commented on fabric-network/test/transaction.js:
```diff
@@ -301,15 +301,15 @@ describe('Transaction', () => {
 		});
 
 		it('sends proposal with long timeout', async () => {
```
With the new behaviour, this test might be better named.
```suggestion
		it('uses endorseTimeout option as proposal timeout', async () => {
```

takayuki-nagai replied:
Thanks for your comment. I'll reflect your advice in the source code.

bestbeforetoday reviewed (changes requested):
Generally looks good. Just a minor test change suggestion.

takayuki-nagai committed:
Define endorseTimeout value in fabric-sdk-node v1.4(Issue #552)

In fabric-sdk-node v1.4, as well as in v2.2, define "endorseTimeout" value and use it instead of "commitTimeout".

Co-Authored-by: Shinsuke Hasegawa <shinsuke.hasegawa.fc@hitachi.com>
Signed-off-by: takayuki-nagai <takayuki.nagai.nu@hitachi.com>

bestbeforetoday reviewed (approved):
Thank you for the contribution! 👍
